trust-quorum/src/connection_manager.rs

```diff
@@ -0,0 +1,705 @@
+        }
+    }
+
+    pub fn status(&self) -> ConnMgrStatus {
+        let connections = self
+            .connecting
+            .iter()
+            .map(|(addr, task_handle)| ConnInfo {
+                state: ConnState::Connecting,
+                addr: *addr,
+                task_id: task_handle.task_id,
+            })
+            .chain(self.accepting.iter().map(|(addr, task_handle)| ConnInfo {
+                state: ConnState::Accepting,
+                addr: *addr,
+                task_id: task_handle.task_id,
+            }))
+            .chain(self.established.iter().map(
+                |(baseboard_id, task_handle)| ConnInfo {
+                    state: ConnState::Established(baseboard_id.clone()),
+                    addr: task_handle.addr(),
+                    task_id: task_handle.task_id,
+                },
+            ))
+            .collect();
+
+        ConnMgrStatus {
+            bootstrap_addrs: self.bootstrap_addrs.clone(),
+            connections,
+            num_conn_tasks: self.join_set.len() as u64,
+            total_tasks_spawned: self.total_tasks_spawned,
+        }
+    }
+
+    pub fn listen_addr(&self) -> SocketAddrV6 {
+        self.listen_addr
+    }
+
+    /// Perform any polling related operations that the connection
+    /// manager must perform concurrently.
+    pub async fn step(
+        &mut self,
+        corpus: Vec<Utf8PathBuf>,
+    ) -> Result<(), AcceptError> {
+        tokio::select! {
+            acceptor = self.server.accept(corpus.clone()) => {
+                self.accept(acceptor?).await?;
+            }
+            Some(res) = self.join_set.join_next_with_id() => {
+                match res {
+                    Ok((task_id, _)) => {
+                        self.on_task_exit(task_id).await;
+                    }
+                    Err(err) => {
+                        error!(self.log, "Connection task panic: {}", err);
+                        self.on_task_exit(err.id()).await;
+                    }
+
+                }
+            }
+            _ = self.reconnect_interval.tick() => {
+                self.reconnect(corpus.clone()).await;
+            }
+        }
+
+        Ok(())
+    }
+
+    pub async fn accept(
+        &mut self,
+        acceptor: SprocketsAcceptor,
+    ) -> Result<(), AcceptError> {
+        let addr = match acceptor.addr() {
+            SocketAddr::V4(addr) => {
+                return Err(AcceptError::Ipv4Accept { addr });
+            }
+            SocketAddr::V6(addr) => addr,
+        };
+        let log = self.log.clone();
+        let (tx, rx) = mpsc::channel(CHANNEL_BOUND);
+        let main_tx = self.main_tx.clone();
+        let abort_handle = self.join_set.spawn(async move {
+            let stream = match acceptor.handshake().await {
+                Ok((stream, _)) => stream,
+
+                Err(err) => {
+                    error!(log, "Failed to accept a connection"; &err);
+                    return ();
```

Comment:
`return ();` feels mildly goofy...i assume this is because otherwise, the return type wouldn't be inferred or something?

Reply:
Nope, this was just me having a brain fart when changing the type from `TaskId` to `()` since the former was no longer needed. Changed to just `return;`